SimPEG/ObjectiveFunction.py

```diff
@@ -0,0 +1,258 @@
+        raise NotImplementedError(
+            "The method deriv has not been implemented for {}".format(
+                self.__class__.__name__
+            )
+        )
+
+    @Utils.timeIt
+    def deriv2(self, x, **kwargs):
+        raise NotImplementedError(
+            "The method _deriv2 has not been implemented for {}".format(
+                self.__class__.__name__
+            )
+        )
+
+    def _test_deriv(self, x=None, num=4, plotIt=False, **kwargs):
+        print('Testing {0!s} Deriv'.format(self.__class__.__name__))
+        if x is None:
+            if self.nP == '*':
+                x = np.random.randn(np.random.randint(1e2, high=1e3))
+            else:
+                x = np.random.randn(self.nP)
+
+        return checkDerivative(
+            lambda m: [self(m), self.deriv(m)], x, num=num, plotIt=plotIt
+        )
+
+    def _test_deriv2(self, x=None, num=4, plotIt=False, **kwargs):
+        print('Testing {0!s} Deriv2'.format(self.__class__.__name__))
+        if x is None:
+            if self.nP == '*':
+                x = np.random.randn(np.random.randint(1e2, high=1e3))
+            else:
+                x = np.random.randn(self.nP)
+
+        return checkDerivative(
+            lambda m: [self.deriv(m), self.deriv2(m)], x, num=num,
+            plotIt=plotIt
+        )
+
+    def test(self, x=None, num=4, plotIt=False, **kwargs):
+        deriv = self._test_deriv(x=x, num=num, **kwargs)
+        deriv2 = self._test_deriv2(x=x, num=num, plotIt=False, **kwargs)
+        return (deriv & deriv2)
+
+    def __add__(self, objfct2):
+
+        if isinstance(objfct2, Utils.Zero):
+            return self
+
+        if not isinstance(objfct2, BaseObjectiveFunction):
+            raise Exception(
+                "Cannot add type {} to an objective function. Only "
+                "ObjectiveFunctions can be added together".format(
+                    objfct2.__class__.__name__
+                )
+            )
+
+        if isinstance(self, ComboObjectiveFunction):
+
+            if isinstance(objfct2, ComboObjectiveFunction):
+                objfctlist = self.objfcts + objfct2.objfcts
+                multipliers = self.multipliers + objfct2.multipliers
+
+            elif isinstance(objfct2, ObjectiveFunction):
+                objfctlist = self.objfcts.append(objfct2)
+                multipliers = self.multipliers.append(1)
+
+        else:
+            if isinstance(objfct2, ComboObjectiveFunction):
+                objfctlist = [self] + objfct2.objfcts
+                multipliers = [1] + objfct2.multipliers
+
+            else:
+                objfctlist = [self, objfct2]
+                multipliers = None
+
+        return ComboObjectiveFunction(
+            objfcts=objfctlist, multipliers=multipliers
+        )
+
+    def __radd__(self, objfct2):
+        return self+objfct2
+
+    def __mul__(self, multiplier):
+        return ComboObjectiveFunction([self], [multiplier])
```

Comment:
there is a check in the `__init__` of the `ComboObjectiveFunction`

![image](https://cloud.githubusercontent.com/assets/6361812/21532482/c9e9c280-cd0d-11e6-885b-7b07152641a7.png)